File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(nums.size());
        res[0] = 1;
        for(int i = 1 ; i < nums.size() ; i++){
            res[i] = nums[i-1]*res[i - 1];
        }
        int p = 1 ;
        for(int i = nums.size() - 2 ; i >=0 ; i--){
            p = p * nums[i+1];
            res[i] = res[i] * p;
        }
        return res;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q = collections.deque()        
        level = []
        q.append(root)
        res = []
        if root :
            while len(q) != 0 :
                level = []
                length = len(q)            
                for i in range(length):
                    first = q.popleft()
                    level.append(first.val) 
                
                    if first.left:
                        q.append(first.left)
                    if first.right:
                        q.append(first.right)
                res.append(level)
        return res

        


